### ISSUE TITLE
nvidia-docker2 package dependency doesn't match on AWS AMIs

### ISSUE THREAD

**Issue by mikecouk** [NVIDIA/nvidia-docker#618]
It seems that even with version pinning enabled, the installation of nvidia-docker2 fails, possibly because the version number of docker is slightly difference once AWS have got their hands on it. 

**Command is ;**

sudo yum install -y nvidia-docker2-2.0.2-1.docker17.06.2.ce.noarch nvidia-container-runtime-1.1.1-1.docker17.06.2.x86_64

**Error is ;**

Error: Package: nvidia-docker2-2.0.2-1.docker17.06.2.ce.noarch (nvidia-docker)
Requires: docker-ce = 17.06.2.ce

**Docker Version ;**

docker -v
Docker version 17.06.2-ce, build 3dfb8343b139d6342acfd9975d7f1068b5b1c3d3

Could this be because the version number has a "period" rather than a "dash" at the end before "ce" ??

Also attached the yum provide output for extra info below ;

_yum provides docker
Loaded plugins: dkms-build-requires, priorities, update-motd, upgrade-helper
libnvidia-container                                                                                                                                                                                      15/15
nvidia-container-runtime                                                                                                                                                                                 18/18
nvidia-docker                                                                                                                                                                                            20/20
2 packages excluded due to repository priority protections
docker-17.03.2ce-1.59.amzn1.x86_64 : Automates deployment of containerized applications
Repo        : amzn-main_

I've tried with 2 different AWS Docker versions and matching nvidia-docker versions , docker 17.06.2 and docker 17.09.1, both have the same error.
AWS AMI is ami-af8013d6 on a p2.xlarge.

**Comment by ramarnat** [comment]
Unfortunately that isnt enough. It also needs an updated libcap
```
git clone https://kernel.googlesource.com/pub/scm/linux/kernel/git/morgan/libcap.git
cd libcap
make
sudo cp libcap/libcap.so.2.25 /lib64
sudo rm /lib64/libcap.so.2
sudo ln -s /lib64/libcap.so.2.25 /lib64/libcap.so.2
```

**Comment by spkaplan** [comment]
Is there a solution for this?

**Comment by ramarnat** [comment]
@spkaplan The way I solved it was to build the above build steps into the docker repos that I use. I believe the maintainers are looking to update the libs

**Comment by spkaplan** [comment]
@ramarnat Thank you!

**Comment by ramarnat** [comment]
@spkaplan I don't know if I can get to it before the middle of march, but I put the init script I use that to make an amazon linux base AMI ECS ready with nvidia. Maybe something you can use in the interim.

**Comment by spkaplan** [comment]
@ramarnat Thank you for providing the init script! I will see if I can get something working with that.

**Comment by flx42** [comment]
We now have packages for the AWS AMIs. Please open a new issue if it doesn't work for you.
